### PR TITLE
Either DecodeFailure or Result, add Argonaut interop

### PR DESCRIPTION
**Pull request by mcgizzle**
Also added scalafmt

### CODE REVIEW

File: circe/src/test/scala/io/github/mcgizzle/circe/CirceDecoderTests.scala [hunk 1]
```diff
@@ -8,26 +8,25 @@ import shapeless.test.illTyped
 class CirceDecoderTests extends FlatSpec with Matchers {
 
   "Decoder" should "provide an implicit Decoder instance given a circe one" in {
-    import io.circe.generic.auto._
+      import io.circe.generic.auto._
```
Comment (jbwheatley) [inline circe/src/test/scala/io/github/mcgizzle/circe/CirceDecoderTests.scala]: why all this whitespace lol

File: argonaut/src/main/scala/io/github/mcgizzle/circe/package.scala [hunk 2]
```diff
@@ -0,0 +1,3 @@
+package io.github.mcgizzle
+
+package object circe extends ArgonautDecoder
```
Comment (jbwheatley) [inline argonaut/src/main/scala/io/github/mcgizzle/circe/package.scala]: shouldn't this package be called argonaut?

**Review by jbwheatley — COMMENTED** [review]
could add a type alias for `Either[DecodeFailure, A]`

File: argonaut/src/main/scala/io/github/mcgizzle/circe/ArgonautDecoder.scala [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package io.github.mcgizzle.circe
+
+import argonaut.{DecodeJson, Json}
+import io.github.mcgizzle.{DecodeFailure, Decoder}
+
+trait ArgonautDecoder {
+  implicit def circeDecoder[A](implicit D: DecodeJson[A]): Decoder[Json, A] =
```
[the remark this answers is not in the document]
Comment (mcgizzle) [reply argonaut/src/main/scala/io/github/mcgizzle/circe/ArgonautDecoder.scala]: silly me

File: circe/src/test/scala/io/github/mcgizzle/circe/CirceDecoderTests.scala (hunk 1 above)
Comment (mcgizzle) [reply circe/src/test/scala/io/github/mcgizzle/circe/CirceDecoderTests.scala]: trying out scalafmt, haven't got the config file setup just right yet it seems

File: argonaut/src/main/scala/io/github/mcgizzle/circe/package.scala (hunk 2 above)
Comment (mcgizzle) [reply argonaut/src/main/scala/io/github/mcgizzle/circe/package.scala]: aghh intelliJ rename was doing untold things

**Review by jbwheatley — APPROVED** [review]
looks good

**Comment by mcgizzle** [comment]
Think I caught all the naming issues, copied circe folder and ran an intelliJ rename which mucked things up.

What are you thinking for the alias ? @jbwheatley

**Pull request merged**